pypeit/coadd2d.py

```diff
@@ -278,49 +356,38 @@ def coadd(self, only_slits=None, interp_dspat=True):
 
         """
 
-        only_slits = [only_slits] if (only_slits is not None and
-                                      isinstance(only_slits, (int, np.int, np.int64, np.int32))) else only_slits
-        # TODO We should be checking the bitmask for the reductions or something here??
-        #  Yes, definitely
-        if only_slits is None:
-            slits = self.stack_dict['slits_list'][0]
-            reduce_bpm = (slits.mask > 0) & (np.invert(slits.bitmask.flagged(
-                slits.mask, flag=slits.bitmask.exclude_for_reducing)))
-            good_slits = np.where(np.invert(reduce_bpm))[0]
-        else:
-            raise NotImplementedError('Slit selection (only_slits) not implemented yet!')
+        self.good_slits = self.good_slitindx(only_slits=only_slits)
 
         coadd_list = []
-        for slit_idx in good_slits:
-            slitord_id = self.stack_dict['slits_list'][0].slitord_id[slit_idx]
-            msgs.info('Performing 2d coadd for slit: {:d}/{:d}'.format(slit_idx, self.nslits - 1))
+        for slit_idx in self.good_slits:
+            msgs.info('Performing 2d coadd for slit: {:d}/{:d}'.format(slit_idx, self.nslits_single - 1))
             ref_trace_stack = self.reference_trace_stack(slit_idx, offsets=self.offsets,
                                                          objid=self.objid_bri)
             thismask_stack = np.abs(self.stack_dict['slitmask_stack'] - self.stack_dict['slits_list'][0].spat_id[slit_idx]) <= self.par['coadd2d']['spat_toler']
 
-            # TODO Can we get rid of this one line simply making the weights returned by parse_weights an
-            # (nslit, nexp) array?
-            # This one line deals with the different weighting strategies between MultiSlit echelle. Otherwise, we
-            # would need to copy this method twice in the subclasses
-            if 'auto_echelle' in self.use_weights:
-                rms_sn, weights = self.optimal_weights(slitord_id, self.objid_bri)
```

Comment:
I'm a but confused bout what happened to this "auto_echelle" functionality. It appears that you have removed it.